src/carriers/unix/UnixSocketCarrier.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (C) 2006-2020 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#include "UnixSocketCarrier.h"
+#include <yarp/conf/filesystem.h>
+
+#include <yarp/os/ConnectionState.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Os.h>
+
+#include "UnixSocketLogComponent.h"
+
+using namespace yarp::os;
+namespace fs = yarp::conf::filesystem;
+
+namespace {
+
+// FIXME: This method should be available somewhere in YARP
+std::string getYARPRuntimeDir()
+{
+    static std::mutex m;
+    std::lock_guard<std::mutex> lock(m);
+
+    static std::string yarp_runtime_dir;
+    bool found = false;
+
+    // If already populated, there is nothing to do
+    if (!yarp_runtime_dir.empty()) {
```

Comment:
Codacy found an issue: [Condition '!yarp_runtime_dir.empty()' is always false](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5914687)

Reply:
For some reason codacy does not realize that `yarp_runtime_dir` is a static variable... dismissing this